packages/office-ui-fabric-react/src/components/DetailsList/DetailsRow.scss

```diff
@@ -12,6 +12,7 @@ $unselectedHoverColor: $ms-color-neutralLighter;
   display: inline-block;
   min-width: 100%;
```

Comment:
As long as you test it :)

I think it should work like so:

Rows are width: 100%
They're rendered in a container that's also inline-block but min-width: 100%
The container is rendered in an overflow hidden or auto DIV, so that if 1 row's contents grow beyond the 100%, all rows can stretch to that width.